lib/src/test/java/vmath/Mat3Test.java

```diff
@@ -51,6 +51,32 @@ void givenNullArray_whenMat3Created_thenThrowsException() {
         assertThrows(IllegalArgumentException.class, () -> new Mat3(a));
     }
 
+    @Test
+    public void givenNaNValues_whenConstructorCalled_thenThrowArithmeticException() {
+        assertThrows(ArithmeticException.class, () -> {
+            new Mat3(1.0f, 2.0f, Float.NaN,
+                    4.0f, 5.0f, 6.0f,
+                    7.0f, 8.0f, 9.0f);
+        });
+    }
+
+    @Test
+    public void givenValidValues_whenConstructorCalled_thenNoExceptionThrown() {
+        var m1 = new Mat3(1.0f, 2.0f, 3.0f,
+                4.0f, 5.0f, 6.0f,
+                7.0f, 8.0f, 9.0f);
+
+        assertEquals(1.0f, m1.get(0, 0));
+        assertEquals(2.0f, m1.get(0, 1));
+        assertEquals(3.0f, m1.get(0, 2));
+        assertEquals(4.0f, m1.get(1, 0));
+        assertEquals(5.0f, m1.get(1, 1));
+        assertEquals(6.0f, m1.get(1, 2));
+        assertEquals(7.0f, m1.get(2, 0));
+        assertEquals(8.0f, m1.get(2, 1));
+        assertEquals(9.0f, m1.get(2, 2));
+    }
```

Comment:
1) Rename the Test Method: Please rename the method to givenArrayArg_whenMat3Created_thenMatrixHasGivenValues. This change aligns with our naming conventions and clearly describes the test's purpose.
2) Array and Object Initialization: Create an array named a of type float[]. Then, initialize a variable m as an instance of Mat3, using the previously created array a.
3) Formatting Consistency: Insert empty lines between each set of three assertEquals statements to maintain consistency with the overall code structure.

Additionally, check the existing Mat4Test to ensure your changes align with the existing code.